File: src/test/java/org/hisp/dhis/rules/models/RuleVariableNewestEventTests.java
package org.hisp.dhis.rules.models;

import org.hisp.dhis.rules.models.RuleValueType;
import org.hisp.dhis.rules.models.RuleVariableNewestEvent;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import static junit.framework.TestCase.fail;
import static org.assertj.core.api.Java6Assertions.assertThat;

@RunWith( JUnit4.class )
public class RuleVariableNewestEventTests
{

        @Test
        public void createShouldThrowOnNullName()
        {
                try
                {
                        RuleVariableNewestEvent.Companion.create( null, "test_dataelement", RuleValueType.TEXT );
                        fail( "NullPointerException is expected, but nothing was thrown" );
                }
                catch ( Exception exception )
                {
                        // noop
                }
        }

        @Test
        public void createShouldThrowOnNullDataElement()
        {
                try
                {
                        RuleVariableNewestEvent.Companion.create( "test_variable", null, RuleValueType.TEXT );
                        fail( "NullPointerException is expected, but nothing was thrown" );
                }
                catch ( Exception exception )
                {
                        // noop
                }
        }

        @Test
        public void createShouldThrowOnNullDataElementType()
        {
                try
                {
                        RuleVariableNewestEvent.Companion.create( "test_variable", "test_dataelement", null );
                        fail( "NullPointerException is expected, but nothing was thrown" );
                }
                catch ( Exception exception )
                {
                        // noop
                }
        }

        @Test
        public void createShouldPropagatePropertiesCorrectly()
        {
                RuleVariableNewestEvent ruleVariableNewestEvent = RuleVariableNewestEvent.Companion.create(
                    "test_variable", "test_dataelement", RuleValueType.NUMERIC );

                assertThat( ruleVariableNewestEvent.getName() ).isEqualTo( "test_variable" );
                assertThat( ruleVariableNewestEvent.getDataElement() ).isEqualTo( "test_dataelement" );
                assertThat( ruleVariableNewestEvent.getDataElementType() ).isEqualTo( RuleValueType.NUMERIC );
        }
}

File: src/test/java/org/hisp/dhis/rules/models/RuleVariableNewestEventTests.kt
package org.hisp.dhis.rules.models

import org.assertj.core.api.Java6Assertions.assertThat
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.JUnit4
import kotlin.test.assertFailsWith

@RunWith(JUnit4::class)
class RuleVariableNewestEventTests {

    @Test
    fun createShouldThrowOnNullName() {
        assertFailsWith<NullPointerException> {
            RuleVariableNewestEvent.create(null!!, "test_dataelement", RuleValueType.TEXT)
        }
    }

    @Test
    fun createShouldThrowOnNullDataElement() {
        assertFailsWith<NullPointerException> {
            RuleVariableNewestEvent.create("test_variable", null!!, RuleValueType.TEXT)
        }
    }

    @Test
    fun createShouldThrowOnNullDataElementType() {
        assertFailsWith<NullPointerException> {
            RuleVariableNewestEvent.create("test_variable", "test_dataelement", null!!)
        }
    }

    @Test
    fun createShouldPropagatePropertiesCorrectly() {
        val ruleVariableNewestEvent = RuleVariableNewestEvent.create(
                "test_variable", "test_dataelement", RuleValueType.NUMERIC)

        assertThat(ruleVariableNewestEvent.name).isEqualTo("test_variable")
        assertThat(ruleVariableNewestEvent.dataElement).isEqualTo("test_dataelement")
        assertThat(ruleVariableNewestEvent.dataElementType).isEqualTo(RuleValueType.NUMERIC)
    }
}
